app/src/main/java/org/schabi/newpipe/VideoItemListFragment.java

```diff
@@ -195,18 +196,31 @@ public void setStreamingService(StreamingService streamingService) {
         this.streamingService = streamingService;
     }
 
-    private void updateListOnResult(SearchResult result, int requestId) {
-        if(requestId == currentRequestId) {
+    public void updateListOnResult(final SearchResult result, int requestId) {
+        if (requestId == currentRequestId) {
             setListShown(true);
             updateList(result.resultList);
-            if(!result.suggestion.isEmpty()) {
-                Toast.makeText(getActivity(),
-                        String.format(getString(R.string.did_you_mean), result.suggestion),
-                        Toast.LENGTH_LONG).show();
-            }
+            if (!result.suggestion.isEmpty())
+                Snackbar.make(getView(), String.format(getString(R.string.did_you_mean, result.suggestion)), Snackbar.LENGTH_LONG)
+                        .setAction(R.string.yes, new View.OnClickListener() {
+                            @Override
+                            public void onClick(View v) {
+                                query = result.suggestion;
+                                updatesearchQuery(query);
+                                Toast.makeText(getActivity(), R.string.search + ": " + String.format(query), Toast.LENGTH_LONG).show();
+                            }
+                        })
+                        .show();
         }
     }
 
+    public void updatesearchQuery(String query) {
```

Comment:
Please write proper camel case, meaning write `updateSearchQuery()` instead of `updatesearchQuery`.